### ISSUE TITLE
Why does animation only work with useState?

### ISSUE THREAD

**Issue by noobvim** [software-mansion/react-native-reanimated#779]
If I change 
**const [X] = useState(new Value(0))**; 
to 
**const X = new Value(0);**
the event will update the Value, but the View will not animate. I can see X updating if I throw a debug('X', X) in onPan, but the Animated.View with styles.box won't move along the x-axis). Why is that? Thanks!!

```javascript
import React, { useRef, useState } from 'react';
import { StyleSheet } from 'react-native';
import { PanGestureHandler, PinchGestureHandler, RotationGestureHandler, State } from 'react-native-gesture-handler';
import Animated from 'react-native-reanimated';

const { 
  add,
  block,
  call,
  concat,
  cond,
  debug,
  event,
  eq,
  greaterThan,
  multiply,
  onChange,
  set,
  timing,
  useCode,
  Value,
} = Animated;

const styles = StyleSheet.create({
  area: {
    position: 'absolute',
    top: 0,
    zIndex: 999,
  },
  box: {
    backgroundColor: 'lightblue',
    height: 200,
    position: 'absolute',
    top: 0,
    width: 200,
  },
});

const Move = () => {
  const rotationRef = useRef();
  const panRef = useRef();
  const pinchRef = useRef();

  const offsetX = new Value(0);
  const offsetY = new Value(0);
  const offsetR = new Value(0);
  const offsetZ = new Value(1);

  const [X] = useState(new Value(0));
  const [Y] = useState(new Value(0));
  const [R] = useState(new Value(0));
  const [Z] = useState(new Value(1));

  const onPan = event([
    {
      nativeEvent: ({ translationX: x, translationY: y, state }) => block([
        set(X, add(x, offsetX)),
        set(Y, add(y, offsetY)),
        cond(
          eq(state, State.END),
          [
            set(offsetX, add(offsetX, x)),
            set(offsetY, add(offsetY, y)),
          ],
        ),
      ]),
    },
  ]);

  const onRotate = event([
    {
      nativeEvent: ({ rotation: r, state }) => block([
        set(R, add(r, offsetR)),
        cond(
          eq(state, State.END), 
          [set(offsetR, add(offsetR, r))]
        ),
      ]),
    },
  ]);

  const onZoom = event([
    {
      nativeEvent: ({ scale: z, state }) => block([
        cond(eq(state, State.ACTIVE), set(Z, multiply(z, offsetZ))),
        cond(eq(state, State.END), [set(offsetZ, multiply(offsetZ, z))]),
      ]),
    },
  ]);

  return (
    <>
      <PanGestureHandler
        minDist={10}
        ref={panRef}
        onGestureEvent={onPan}
        onHandlerStateChange={onPan}
        simultaneousHandlers={[rotationRef, pinchRef]}
      >
        <Animated.View style={styles.area}>
          <PinchGestureHandler
            onGestureEvent={onZoom}
            onHandlerStateChange={onZoom}
            ref={pinchRef}
            simultaneousHandlers={[rotationRef, panRef]}
          >
            <Animated.View>
              <RotationGestureHandler
                onGestureEvent={onRotate}
                onHandlerStateChange={onRotate}
                ref={rotationRef}
                simultaneousHandlers={[pinchRef, panRef]}
              >
                <Animated.View
                  style={[
                    styles.box,
                    {
                      transform: [
                        { translateX: X },
                        { translateY: Y },
                        { rotate: concat(R, 'rad') },
                        { scale: Z },
                      ],
                    }
                ]}
                />
              </RotationGestureHandler>
            </Animated.View>
          </PinchGestureHandler>
        </Animated.View>
      </PanGestureHandler>
    </>
  );
};
```

**Comment by hbarylskyi** [comment]
@noobvim were you able to find a solution? I seem to have the problem with components not animating though values are set and logged to console. Spent whole day on this.

```
const onGestureEvent = event([
    {
      nativeEvent: ({ translationX: x, translationY: y, state: _state }) => {
        return block([
          set(gestureX, x),
          set(gestureY, y),
          set(state, _state),

          debug('STATE!', state),
        ]);
      },
    },
  ]);```

**Comment by jakub-gonet** [comment]
@hbarylskyi, do you use expo or RN? It may be logged in the native console though - try checking logcat / using XCode


Great reply by @DrRefactor [here](https://github.com/software-mansion/react-native-reanimated/issues/780#issuecomment-628724440) should clarify some confusion.

If you still will be experiencing an issue with animation not working you can make a new issue.

@noobvim, can we close this issue?

**Comment by hbarylskyi** [comment]
@jakub-gonet thanks for your help. I was able to resolve my problem. It wasn’t related to setState (because components render only once), it was related to my implementation.

**Comment by jakub-gonet** [comment]
Closing as inactive.
